src/include/daos_srv/vos.h

```diff
@@ -1514,6 +1521,66 @@ vos_pool_needs_checkpoint(daos_handle_t poh);
 int
 vos_pool_checkpoint(daos_handle_t poh);
 
+typedef void (*bind_cpu_fn_t)(int tgt_id);
+/**
+ * Recreate pool vos/rdb file on storage_path
+ *
+ * \param[in] pool_uuid		Pool uuid
+ * \param[in] scm_size		Per vos file size
+ * \param[in] tgt_nr		vos files number
+ * \param[in] rdb_blob_sz	rdb file size
+ * \param[in] storage_path	Base path for store vos/rdb files
+ * \param[in] bind_cpu_fn	Bind a separate cpu to each vos file allocation
+ */
+int
+vos_pool_recreate_tgt(uuid_t pool_uuid, daos_size_t scm_size, int tgt_nr, daos_size_t rdb_blob_sz,
+		      const char *storage_path, bind_cpu_fn_t bind_cpu_fn);
+
+/**
+ * Generate the vos file path.
+ * Rule: '/${dir}/${pool_uuid}/${fname}-${idx}'
+ * e.g:  '/mnt/daos/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/vos-0'
+ *
+ * \param[in] pool_uuid		Pool uuid
+ * \param[in] dir			Base path for scm mountpoint
+ * \param[in] fname			Vos/rdb name
+ * \param[in] idx			Vos file index
+ * \param[in] fpath			Save the generated path
+ */
+int
+vos_path_gen(const uuid_t pool_uuid, const char *dir, const char *fname, int *idx, char **fpath);
+
+/**
+ * Sync cache
+ */
+int
+vos_dir_fsync(const char *path);
+
+/**
+ * Parallel recreate vos file
+ * \param[in] uuid				Pool uuid
+ * \param[in] scm_size			Per vos file size
+ * \param[in] tgt_nr			Vos file counts
+ * \param[in] cancel_pending	if true, preallocate will abort
+ * \param[in] storage_path		Base path for store vos/rdb files
+ * \param[in] bind_cpu_fn		Usually 'dss_bind_to_xstream_cpuset', this func can't be
+ * call in vos, so specify pointer
```

Comment:
IMHO too many implementation details.

```suggestion
 * \param[in] bind_cpu_fn		e.g. `dss_bind_to_xstream_cpuset`
```